src/applyMiddleware.js

```diff
@@ -24,7 +24,7 @@ export default function applyMiddleware(...middlewares) {
 
     var middlewareAPI = {
       getState: store.getState,
-      dispatch: (action) => dispatch(action)
+      dispatch: (action) => store.dispatch(action)
```

Comment:
With this change, the assignment above is not needed:
```
var dispatch = store.dispatch
```
-- only the variable declaration for the following line:
```
dispatch = compose(...chain)(store.dispatch)
```